conductor/src/main.rs

```diff
@@ -670,36 +680,63 @@ async fn init_cloud_perms(
+    if !is_cloud_formation && cloud_provider != "azure" {
         return Ok(());
     }
 
-    create_cloudformation(
-        aws_region.clone(),
-        backup_archive_bucket.clone(),
-        read_msg.message.namespace.clone(),
-        read_msg.message.backups_read_path.clone(),
-        read_msg.message.backups_write_path.clone(),
-        cf_template_bucket,
-    )
-    .await?;
-
-    // Lookup the CloudFormation stack's role ARN
-    let role_arn = lookup_role_arn(aws_region, &read_msg.message.namespace).await?;
-
-    info!("{}: Adding backup configuration to spec", read_msg.msg_id);
-    // Format ServiceAccountTemplate spec in CoreDBSpec
-    use std::collections::BTreeMap;
-    let mut annotations: BTreeMap<String, String> = BTreeMap::new();
-    annotations.insert("eks.amazonaws.com/role-arn".to_string(), role_arn.clone());
-    let service_account_template = ServiceAccountTemplate {
-        metadata: Some(ObjectMeta {
-            annotations: Some(annotations),
-            ..ObjectMeta::default()
-        }),
-    };
+    // These vary based on cloud provider
+    let mut service_account_template = ServiceAccountTemplate::default();
+    let mut destination_path = "".to_string();
+
+    if cloud_provider == "aws" {
+        let write_path = read_msg
+            .message
+            .backups_write_path
+            .clone()
+            .unwrap_or(format!("v2/{}", read_msg.message.namespace));
+
+        create_cloudformation(
+            aws_region.clone(),
+            backup_archive_bucket.clone(),
+            read_msg.message.namespace.clone(),
+            read_msg.message.backups_read_path.clone(),
+            read_msg.message.backups_write_path.clone(),
+            cf_template_bucket,
+        )
+        .await?;
+
+        // Lookup the CloudFormation stack's role ARN
+        let role_arn = lookup_role_arn(aws_region, &read_msg.message.namespace).await?;
+
+        info!("{}: Adding backup configuration to spec", read_msg.msg_id);
+        // Format ServiceAccountTemplate spec in CoreDBSpec
+        use std::collections::BTreeMap;
+        let mut annotations: BTreeMap<String, String> = BTreeMap::new();
+        annotations.insert("eks.amazonaws.com/role-arn".to_string(), role_arn.clone());
+        service_account_template = ServiceAccountTemplate {
+            metadata: Some(ObjectMeta {
+                annotations: Some(annotations),
+                ..ObjectMeta::default()
+            }),
+        };
+        destination_path = format!("s3://{}/{}", backup_archive_bucket, write_path);
+    }
+
+    if cloud_provider == "azure" {
+        let write_path = read_msg
+            .message
+            .backups_write_path
+            .clone()
+            .unwrap_or(format!("{}", read_msg.message.namespace));
+        destination_path = format!(
+            "https://{}.blob.core.windows.net/{}/{}",
+            azure_storage_account, backup_archive_bucket, write_path
+        );
+    }
```

Comment:
I think this may need a step to provision in Azure the permissions for the workload required for this instance to access the bucket where its backups go

Reply:
:+1: Yep we will need that